app/services/foreman_ansible/inventory_creator.rb

```diff
@@ -8,11 +8,42 @@ def initialize(hosts)
       @hosts = hosts
     end
 
-    def tempfile
-      tempfile = Tempfile.new("foreman-#{SecureRandom.uuid}-inventory")
-      tempfile.write(hosts.map(&:fqdn).join('\n'))
-      tempfile.close
-      tempfile
+    def json
+      { 'all' => { 'hosts' => hosts.map(&:fqdn) },
+        '_meta' => { 'hostvars' => host_vars } }
```

Comment:
The contents of the `json['_meta']['hostvars'][fqdn]` gets assigned to the top level, so `json['_meta']['hostvars']['localhost']['foreman']` can be accessed just as `foreman` in the playbook

See [ansible docs](http://docs.ansible.com/ansible/developing_inventory.html#tuning-the-external-inventory-script) for details on the `_meta` magic